File: kiv9908/week_3/168p.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        
        for (int i = 0; i < moves.length; i++) {
            int temp = 0;
            for (int j = 0; j < board.length; j++) {
                if (board[j][moves[i] - 1] != 0) {
                    temp = board[j][moves[i] - 1];
                    board[j][moves[i] - 1] = 0;
                    break;
                }
            }
            if (temp != 0) {
                if (stack.isEmpty()) {
                    stack.push(temp);
                } else {
                    if (stack.peek() == temp) {
                        int temp2 = stack.pop();
                        answer += 2;
                    } else {
                        stack.push(temp);
                    }
                }
            }

        }
        return answer;
    }
}

File: jane-kim-279/week_3/168p.py
def solution(board, moves):
    answer = 0
    N = len(board)
    cart = []
    
    def check_cart(cart):
        if len(cart) >= 2 and cart[-1] == cart[-2]:
            cart.pop()
            cart.pop()
            return True
        else:
            return False
    
    for move in moves:
        for i in range(N):
            if board[i][move-1] != 0:
                cart.append(board[i][move-1])
                board[i][move-1] = 0
                if check_cart(cart):
                    answer += 2
                break
    return answer
